src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c

```diff
@@ -618,6 +618,36 @@ tl_data2berval (krb5_tl_data *in, struct berval **out)
     return 0;
 }
 
+static krb5_error_code
+update_ldap_mod_auth_ind(krb5_context context, krb5_db_entry *entry,
+                         LDAPMod ***mods)
+{
+    int i = 0;
+    krb5_error_code ret;
+    char *auth_ind = NULL;
+    char *strval[10] = {};
+    char *ai, *ai_save = NULL;
+    int sv_num = sizeof(strval) / sizeof(*strval);
+
+    ret = krb5_dbe_get_string(context, entry, KRB5_KDB_SK_REQUIRE_AUTH,
+                              &auth_ind);
+    if (auth_ind == NULL || ret != 0)
+        goto cleanup;
+
+    ai = strtok_r(auth_ind, " ", &ai_save);
+    while (ai != NULL && i < sv_num) {
+        strval[i++] = ai;
+        ai = strtok_r(NULL, " ", &ai_save);
+    }
+
+    ret = krb5_add_str_mem_ldap_mod(mods, "krbPrincipalAuthInd",
+                                    LDAP_MOD_REPLACE, strval);
+
+cleanup:
+    free(auth_ind);
```

Comment:
Use krb5_dbe_free_string() instead of free() for this, per the contract of krb5_dbe_get_string().  (It doesn't matter in practice since the whole tree is compiled with the same malloc/free, but it's good hygiene.)